res_users_password_security/controllers/main.py

```
@@ -0,0 +1,90 @@
+
+import operator
+
+from openerp import http
+from openerp.http import request
+from openerp.addons.auth_signup.controllers.main import AuthSignupHome
+from openerp.addons.web.controllers.main import ensure_db, Session
+
+from ..exceptions import PassError
+
+
+class PasswordSecuritySession(Session):
+
+    @http.route()
+    def change_password(self, fields):
+        new_password = operator.itemgetter('new_password')(
+            dict(map(operator.itemgetter('name', 'value'), fields))
+        )
+        user_id = request.env.user
+        user_id.check_password(new_password)
+        return super(PasswordSecuritySession, self).change_password(fields)
+
+
+class PasswordSecurityHome(AuthSignupHome):
+
+    def do_signup(self, qcontext):
+        password = qcontext.get('password')
+        user_id = request.env.user
+        user_id.check_password(password)
+        return super(PasswordSecurityHome, self).do_signup(qcontext)
+
+    @http.route()
+    def web_login(self, *args, **kw):
+        ensure_db()
+        response = super(PasswordSecurityHome, self).web_login(*args, **kw)
+        if not request.httprequest.method == 'POST':
+            return response
+        uid = request.session.authenticate(
+            request.session.db,
+            request.params['login'],
+            request.params['password']
+        )
+        if not uid:
+            return response
+        users_obj = request.env['res.users'].sudo()
+        user_id = users_obj.browse(request.uid)
+        if not user_id._password_has_expired():
+            return response
+        user_id.action_expire_password()
+        redirect = user_id.partner_id.signup_url
+        return http.redirect_with_hash(redirect)
+
+    @http.route()
+    def web_auth_signup(self, *args, **kw):
+        try:
+            return super(PasswordSecurityHome, self).web_auth_signup(
+                *args, **kw
+            )
+        except PassError, e:
```

Could you use `except PassError as e:` instead, please?

---

More info: https://www.python.org/dev/peps/pep-3110